doc/source/data/shuffling-data.rst

```diff
@@ -0,0 +1,225 @@
+.. _shuffling_data:
+
+==============
+Shuffling Data
+==============
+
+When consuming or iterating over Ray :class:`Datasets <ray.data.dataset.Dataset>`, it can be useful to
+shuffle or randomize the order of data (for example, randomizing data ingest order during ML training). 
+This guide shows several different methods of shuffling data with Ray Data and their respective trade-offs.
+
+Types of shuffling
+==================
+
+Ray Data provides several different options for shuffling data, trading off the granularity of shuffle
+control with memory consumption and runtime. The options below are listed in increasing order of 
+resource consumption and runtime; choose the most appropriate method for your use case.
+
+.. _shuffling_file_order:
+
+Shuffle the ordering of files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To randomly shuffle the ordering of input files before reading, call a function like
+:func:`~ray.data.read_images` and specify ``shuffle="files"``. This randomly assigns
+input files to workers for reading.
+
+.. testcode::
+
+    import ray
+
+    ds = ray.data.read_images(
+        "s3://anonymous@ray-example-data/image-datasets/simple",
+        shuffle="files",
+    )
+
+.. tip::
```

Comment:
This didn't really feel like a tip to me. IMO, it might be better to make this regular text here and in the other sections. In general, I think we should try to use admonitions sparingly.